src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -228,10 +228,27 @@ public void resetDrivingModes() {
         m_precisionMode = false;
     }
 
-    // toggle whether driving is field-centric
-    public void toggleFieldCentric() {
-        m_fieldCentric = !m_fieldCentric;
-    }
+        // toggle whether driving is field-centric
+	public void toggleFieldCentric() {
+		m_fieldCentric = !m_fieldCentric;
+	}
+
+    // toggle precision mode for driving
+	public void togglePrecisionMode() {
+		m_precisionMode = !m_precisionMode;
+		m_maxVelocity = m_precisionMode ? MAX_VELOCITY_PRECISION_MODE : MAX_VELOCITY_METERS_PER_SECOND;
+		m_maxAngularVelocity = m_precisionMode ? MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND_PRECISION_MODE
+				: MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND;
+	}
+
+    // lock wheels in x position to resist pushing
+	public void lockWheels() {
+        double lockDegrees = Math.toRadians(45);
```

Comment:
Since the actual value stored in this variable is in radians, the variable name should be lockRadians

Reply:
Still needs fixing